src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -937,6 +933,12 @@ abstract class ManagedStrategy(
             if (owner == null && atomicMethodDescriptor == null && guarantee == null) { // static method
                 LincheckJavaAgent.ensureClassHierarchyIsTransformed(className.canonicalClassName)
             }
+            if (atomicMethodDescriptor != null && atomicMethodDescriptor.kind.isSetter) {
```

Comment:
This code becomes more and more complicated. I would suggest splitting the logical blocks of this method with comments explaining what each of these logical blocks does.

Reply:
Done